presto-main/src/main/java/com/facebook/presto/execution/scheduler/NodeScheduler.java

```diff
@@ -263,8 +264,11 @@ private Supplier<NodeMap> createNodeMapSupplier(ConnectorId connectorId)
                     .map(InternalNode::getNodeIdentifier)
                     .collect(toImmutableSet());
 
-            int weight = (int) ceil(1.0 * minVirtualNodeCount / activeNodes.size());
-            ConsistentHashingNodeProvider consistentHashingNodeProvider = ConsistentHashingNodeProvider.create(activeNodes, weight);
+            Optional<ConsistentHashingNodeProvider> consistentHashingNodeProvider = Optional.empty();
+            if (nodeSelectionHashStrategy == nodeSelectionHashStrategy.CONSISTENT_HASHING) {
+                int weight = (int) ceil(1.0 * minVirtualNodeCount / activeNodes.size());
+                consistentHashingNodeProvider = ConsistentHashingNodeProvider.create(activeNodes, weight);
```

Comment:
lets change this to 
`consistentHashingNodeProvider = Optional.of(ConsistentHashingNodeProvider.create(activeNodes, weight));`

Reply:
Done.